app/models/ethnic_church_builder.rb

```diff
@@ -0,0 +1,49 @@
+class EthnicChurchBuilder
+  class << self
+    def instantiate(params)
+      @params = params
+
+      @ec = EthnicChurch.new(ethnic_church_params)
+
+      @ec.languages = languages
+      @ec.country = country
+      @ec.religious_background = religious_background
+      @ec.build_address(address_params)
+      @ec.build_note(note_params)
+
+      @ec
+    end
+
+    private
+
+    def languages
+      @params[:language][:name].reject(&:blank?).map do |lang_name|
```

Comment:
One thing i noticed is that in your EthnicChurch model u override `lanugages=` I would make that a validation instead.

Reply:
So, I'm actually using the default `#languages=` method as I removed my custom implementation in this pr. 

The `reject(&:blank?)` was intended to handle a blank string that rails multiselects add to the params (due to the way it adds an extra hidden field). See for example https://stackoverflow.com/questions/8929230/why-is-the-first-element-always-blank-in-my-rails-multi-select-using-an-embedde#8939205

Do you think there is a better place or a better way to remove this blank?

Reply:
Whoops. Didnt see that u removed that. I think this is ok but i would add a validation to the language model that says does not allow null/empty string. Also make it so the front end can not send u a blank string. That maye require some js fun or if u are using the rails helper i believe there is an option u can pass in.